editor/src/components/canvas/dom-walker.ts

```diff
@@ -1045,9 +1038,58 @@ function getSpecialMeasurements(
     fontWeight,
     fontStyle,
     textDecorationLine,
+    textBounds,
   )
 }
 
+function elementContainsOnlyText(element: HTMLElement): boolean {
+  if (element.childNodes.length === 0) {
+    return false
+  }
+  for (const node of element.childNodes) {
+    const isForText =
+      node.nodeType === Node.TEXT_NODE ||
+      (node.nodeType === Node.ELEMENT_NODE && node.nodeName === 'BR')
+    if (!isForText) {
+      return false
+    }
+  }
+  return true
+}
+
+function getTextBounds(element: HTMLElement, scale: number): DOMRect | null {
+  const range = document.createRange()
+  range.selectNodeContents(element)
+  if (range.getBoundingClientRect == null) {
+    // for some reason, this can be undefined (in tests?)
```

Comment:
In jest tests only?

Reply:
apparently so

Reply:
That's likely because they run with a fake DOM, which doesn't support everything a regular browser supports.